### PR TITLE
add back loop feature.

### PR DESCRIPTION
**Pull request by erwinmombay**
need to create a new one cause of private -> public switch.

need to finish documentation

TODOS:
- [x] loop on swipe
- [x] loop on button click

### CODE REVIEW

**Comment by erwinmombay** [comment]
@dvoytenko can you check if carousel integration tests still fails for you locally?

**Comment by erwinmombay** [comment]
@dvoytenko this is ready for review.

File: extensions/amp-carousel/0.1/slides.js [hunk 1]
```diff
@@ -300,11 +311,35 @@ export class AmpSlides extends BaseCarousel {
...
+   * @param {number} length length of the vector.
+   */
+  static getRelativeIndex(index, step, length) {
+    assert(index >= 0 && index < length, 'index passed to getRelativeIndex ' +
+        'must be within the length exclusive. Value: ' + index);
+    let nextItemIndex = index + step;
+    if (nextItemIndex < 0) {
+      return length + (nextItemIndex % length);
+    }
+    return nextItemIndex % length;
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/slides.js]: `(index + step + length) % length` is a bit shorter

**Comment by dvoytenko** [comment]
LGTM with a couple of comments.

File: extensions/amp-carousel/0.1/slides.js (hunk 1 above)
Comment (erwinmombay) [inline extensions/amp-carousel/0.1/slides.js]: done.

**Pull request merged**